build.sh

```diff
@@ -28,80 +185,80 @@ function main
-      msg "Using compiler: clang++/clang"
-      cxx="clang++"
-      cc="clang"
-    elif command -v g++ >/dev/null; then
-      msg "Using compiler: g++/gcc"
-      cxx="g++"
-      cc="gcc"
-    fi
-  else
-      cxx="g++"
-      cc="gcc"
-  fi
+  set_build_opts
 
   msg "Executing cmake command"
   cmake                                       \
-    -DCMAKE_C_COMPILER="${cc}"                \
-    -DCMAKE_CXX_COMPILER="${cxx}"             \
-    -DENABLE_ALSA:BOOL="${enable_alsa}"       \
-    -DENABLE_PULSEAUDIO:BOOL="${enable_pulseaudio}"\
-    -DENABLE_I3:BOOL="${enable_i3}"           \
-    -DENABLE_MPD:BOOL="${enable_mpd}"         \
-    -DENABLE_NETWORK:BOOL="${enable_network}" \
-    -DENABLE_CURL:BOOL="${enable_curl}"       \
-    -DBUILD_IPC_MSG:BOOL="${build_ipc_msg}"   \
+    -DCMAKE_C_COMPILER="${CC}"                \
+    -DCMAKE_CXX_COMPILER="${CXX}"             \
+    -DENABLE_ALSA:BOOL="${ENABLE_ALSA}"       \
+    -DENABLE_PULSEAUDIO:BOOL="${ENABLE_PULSEAUDIO}"\
+    -DENABLE_I3:BOOL="${ENABLE_I3}"           \
+    -DENABLE_MPD:BOOL="${ENABLE_MPD}"         \
+    -DENABLE_NETWORK:BOOL="${ENABLE_NETWORK}" \
+    -DENABLE_CURL:BOOL="${ENABLE_CURL}"       \
+    -DBUILD_IPC_MSG:BOOL="${ENABLE_IPC_MSG}"   \
     .. || msg_err "Failed to generate build... read output to get a hint of what went wrong"
 
   msg "Building project"
   make || msg_err "Failed to build project"
-
-  read -r -p "$(msg "Execute 'sudo make install'? [Y/n] ")" -n 1 p && echo
-  [[ "${p^^}" == "Y" ]] && {
-    sudo make install || msg_err "Failed to install executables..."
-  }
-
-  read -r -p "$(msg "Install example configuration? [Y/n]: ")" -n 1 p && echo
-  [[ "${p^^}" == "Y" ]] && {
-    make userconfig || msg_err "Failed to install user configuration..."
-  }
-
+  install
   msg "Build complete!"
 
   exit 0
 }
 
-main "$@"
+
+#################
+###### Entry
+#################
+while [[ "$1" == -* ]]; do
+  case "$1" in
+    -3|--i3)
+      ENABLE_I3=ON; shift ;;
+    -a|--alsa)
+      ENABLE_ALSA=ON; shift ;;
+    -p|--pulse-audio)
+      ENABLE_PULSEAUDIO=ON; shift ;;
+    -n|--network)
+      ENABLE_NETWORK=ON; shift ;;
+    -m|--mpd)
+      ENABLE_MPD=ON; shift ;;
+    -c|--curl)
+      ENABLE_CURL=ON; shift ;;
+    -i|--ipc)
+      ENABLE_IPC_MSG=ON; shift ;;
+    --all-features)
+      ENABLE_I3=ON
+      ENABLE_ALSA=ON
+      ENABLE_PULSEAUDIO=ON
+      ENABLE_NETWORK=ON
+      ENABLE_MPD=ON
+      ENABLE_CURL=ON
+      ENABLE_IPC_MSG=ON
+      shift ;;
+
+    -g|--gcc)
+      USE_GCC=ON; shift ;;
+    -f)
+      REMOVE_BUILD_DIR=ON; shift ;;
+    -I|--no-install)
+      INSTALL=OFF; shift ;;
+    -C|--no-install-config)
+      INSTALL_CONF=OFF; shift ;;
+    -A|--auto)
+      AUTO=ON; shift ;;
+    -h|--help)
+      usage
+      exit 0
+      ;;
+    --) shift; break ;;
+    *)
+      usage
+      [[ "$1" =~ ^-[0-9a-zA-Z]{2,}$ ]] && msg_err "don't combine options: ie do [-c -i] instead of [-ci]" || msg_err "unknown option [$1]"
```

Comment:
Probably would be nicer if this was supported, `getopt` would work for this.

Reply:
But then we lose long format; arg parsing has never been too straight-forward with bash. I'd rather leave as-is - not what users might expect, but at least error message explicitly instructs not to bundle opts.